eox_nelp/pearson_vue/pipeline.py

```diff
@@ -364,18 +385,81 @@ def build_ead_request(
     Returns:
         dict: dict with ead_request dict
     """
-    ead_request = {
-        "@clientAuthorizationID": exam_metadata["client_authorization_id"],
-        "@clientID": getattr(settings, "PEARSON_RTI_WSDL_CLIENT_ID"),
-        "@authorizationTransactionType": transaction_type,
-        "clientCandidateID": f'NELC{profile_metadata["anonymous_user_id"]}',
-        "examAuthorizationCount": exam_metadata["exam_authorization_count"],
-        "examSeriesCode": exam_metadata["exam_series_code"],
-        "eligibilityApptDateFirst": exam_metadata["eligibility_appt_date_first"],
-        "eligibilityApptDateLast": exam_metadata["eligibility_appt_date_last"],
-        "lastUpdate": timezone.now().strftime("%Y/%m/%d %H:%M:%S GMT"),
-    }
+    try:
+        ead_request = {
+            "@clientAuthorizationID": exam_metadata["client_authorization_id"],
+            "@clientID": getattr(settings, "PEARSON_RTI_WSDL_CLIENT_ID"),
+            "@authorizationTransactionType": transaction_type,
+            "clientCandidateID": f'NELC{profile_metadata["anonymous_user_id"]}',
+            "examAuthorizationCount": exam_metadata["exam_authorization_count"],
+            "examSeriesCode": exam_metadata["exam_series_code"],
+            "eligibilityApptDateFirst": exam_metadata["eligibility_appt_date_first"],
+            "eligibilityApptDateLast": exam_metadata["eligibility_appt_date_last"],
+            "lastUpdate": timezone.now().strftime("%Y/%m/%d %H:%M:%S GMT"),
+        }
+    except KeyError as exc:
+        raise PearsonKeyError("EAD", str(exc)) from exc
+    except AttributeError as a_exc:
+        raise PearsonAttributeError("EAD", str(a_exc)) from a_exc
 
     return {
         "ead_request": ead_request
     }
+
+
+def audit_pearson_error(*args, **kwargs):
```

Comment:
I have conflict with this pipe,  I think it's too general and this shouldn't raise a ValueError I think this should raise the same exception that started the `rti_error_handler_task` you are also passing the whole kwargs and args  that could have sensitive data

Reply:
I changed the exception raised. No way, I think that the data audited is similar to the rti import data.
So in terms in sensitive in my opinion is not much difference from [this](https://github.com/eduNEXT/eox-nelp/pull/182#discussion_r1649294134)
Also you could compare  the `audit_pearson_error` [eg](https://edunext.stage.nelp.gov.sa/admin/eox_audit_model/auditmodel/32003/change/)